### PR TITLE
Implement base Collective Policy structure and "Expense Author cannot Approve" policy

### PR DESCRIPTION
**Pull request by kewitz**
Resolves https://github.com/opencollective/opencollective/issues/5184

### CODE REVIEW

File: server/graphql/common/expenses.ts [hunk 1]
```diff
@@ -300,6 +301,16 @@ export const canApprove: ExpensePermissionEvaluator = async (req, expense, optio
     }
     return false;
   } else {
+    const collective = expense.collective || (await models.Collective.findByPk(expense.CollectiveId));
+    if (collective.hasPolicy(POLICIES.EXPENSE_AUTHOR_CANNOT_APPROVE) && req.remoteUser.id === expense.UserId) {
```
Comment (Betree) [inline server/graphql/common/expenses.ts:305]: We must try to stick with loaders as much as we can in these permissions evaluators. This one is called in the expenses list, a loader will prevent calling 15x the DB

```suggestion
    expense.collective = expense.collective || (await req.loaders.Collective.byId.load(expense.CollectiveId));
    if (expense.collective?.hasPolicy(POLICIES.EXPENSE_AUTHOR_CANNOT_APPROVE) && req.remoteUser.id === expense.UserId) {
```

File: server/models/Collective.js [hunk 2]
```diff
@@ -3184,6 +3186,20 @@ function defineModel() {
...
+      }
+    }
+
+    return this.update({ data: { ...this.data, policies: uniq(policies) } });
+  };
+
+  Collective.prototype.hasPolicy = function (policy) {
+    return Boolean(this.data?.policies?.includes(policy));
+  };
+
```
Comment (Betree) [inline server/models/Collective.js:3202]: This is great!

**Review by Betree — COMMENTED** [review]
I love the way you've implemented this generic policy flag at the collective level, that's a simple and solid layer for implementing new things in the future!

File: server/graphql/common/expenses.ts (hunk 1 above)
Comment (kewitz) [reply server/graphql/common/expenses.ts:305]: Perfect! I feel like I always forget about loaders and you're always catching that. 😂

File: server/models/Collective.js (hunk 2 above)
Comment (kewitz) [reply server/models/Collective.js:3202]: Thanks. I added the minimum amount of methods, this can change a bit in the future.

**Pull request merged**